modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -117,8 +117,22 @@ const FrontendDataSet = ({
 	const [total, setTotal] = useState(0);
 
 	const getInitialViewsState = () => {
+		const customInternalViews =
+			customRenderers?.view?.map((customRenderer) => ({
```

Comment:
if we add a new `view` prop in `customRenderers`, then we shall reflect it in the [types ](https://github.com/liferay/liferay-portal/blob/1d910dc1abe4dcf8f763450d1fc472eaf1c124a1/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSetContext.tsx#L17) as well

Reply:
Nice catch! I would honestly love us to dedicate a bit of time for migration to TS. At least for the main file, `FrontendDataSet.js`. It's just too big to do it as a part of some other task.